cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -89,7 +89,8 @@ public void suspendWorkflowRun(HttpRequest request, final HttpResponder responde
       Id.Program id = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowName);
       ProgramRuntimeService.RuntimeInfo runtimeInfo = runtimeService.list(id).get(RunIds.fromString(runId));
       if (runtimeInfo == null) {
-        sendInvalidResponse(responder, id);
+        responder.sendString(HttpResponseStatus.NOT_FOUND, String.format("Runtime information for run-id %s not found",
```

Comment:
Yeah, we should throw NotFoundException instead. HttpExceptionHandler catches the exception and produces a response: https://github.com/caskdata/cdap/blob/develop/cdap-common/src/main/java/co/cask/cdap/common/exception/HttpExceptionHandler.java#L35